pkg/controller/sparkapplication/controller.go

```diff
@@ -1001,3 +1006,19 @@ func (c *Controller) hasApplicationExpired(app *v1beta2.SparkApplication) bool {
 
 	return false
 }
+
+// Clean up batch scheduling config if applicable.
+func (c *Controller) cleanUpBatchScheduling(oldApp, newApp *v1beta2.SparkApplication) error {
```

Comment:
Please rename this to `cleanUpOnTermination`, which sounds more generic and can be extended later to handle deletion of other resources, e.g., the UI service. Please update the comments also.

Reply:
fix it